src/v2/ion/responseTransformer.js

```diff
@@ -170,31 +170,31 @@ const convertRedirectIdPToSuccessRedirectIffOneIdp = (settings, result, lastResu
   }
 };
 
-
 /**
- * API reuses `redirect-idp` remeditaion form for IdP Authenticator for both verify and enroll.
- * Hence IdP Authenticator becomes outlier comparing with other Authenticators in terms of
+ * API reuses `redirect-idp` remediation form for PIV IdP and IdP Authenticator.
+ * IdP Authenticator becomes outlier comparing with other Authenticators in terms of
  * using `challenge-authenticator` and `enroll-authenticator` remediation form.
+ * The UX for PIV IdP is different from other idps in terms of the PIV
+ * instructions view that needs to be rendered before we redirect to mtls.
  *
- * This function change `redirect-idp` to `challenge-authenticator` or `enroll-authenticator`
- * for IdP Authenticator.
+ * This function changes `redirect-idp` to `challenge-authenticator` or `enroll-authenticator`
+ * for IdP Authenticator and changes `redirect-idp` to `piv-idp` for PIV IdP.
  */
-const modifyFormNameForIdPAuthenticator = result => {
+const modifyFormNameForIdP = result => {
   if (Array.isArray(result.remediations)) {
-    const idpAuthenticator = result.remediations.filter(
-      remediation => {
-        return remediation.name === RemediationForms.REDIRECT_IDP
-          && remediation?.relatesTo?.value?.key === 'external_idp';
+    result.remediations.forEach(remediation => {
+      if (remediation.name === RemediationForms.REDIRECT_IDP &&
+          remediation?.relatesTo?.value?.key === 'external_idp') {
```

Comment:
nit: can we also add `external_idp` to the `IDP_FORM_TYPE` constant?

Reply:
This is not IdP type though.. seems to be authenticator key. I can change it to use authenticator key constant 👍